community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/TransactionStateMachineSPI.java

```diff
@@ -64,7 +65,7 @@
     TransactionStateMachineSPI( GraphDatabaseAPI db,
                                 ThreadToStatementContextBridge txBridge,
                                 QueryExecutionEngine queryExecutionEngine,
-                                TransactionIdStore transactionIdStoreSupplier,
+                                Supplier<TransactionIdStore> transactionIdStoreSupplier,
```

Comment:
Could we maybe remove this param and take `transactionIdStoreSupplier` out of `GraphDatabaseAPI db`?
This is not very important but will reduce amount of constructor arguments.